packages/date-picker/src/DatePicker/DatePickerComponent/DatePickerComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  forwardRef,
+  KeyboardEventHandler,
+  TransitionEventHandler,
+  useEffect,
+  useRef,
+} from 'react';
+import { ExitHandler } from 'react-transition-group/Transition';
+
+import { useBackdropClick, useForwardedRef } from '@leafygreen-ui/hooks';
+import { keyMap } from '@leafygreen-ui/lib';
+
+import { useDatePickerContext } from '../../shared/components/DatePickerContext';
+import { DatePickerInput } from '../DatePickerInput';
+import { DatePickerMenu } from '../DatePickerMenu';
+import { useSingleDateContext } from '../SingleDateContext';
+
+import { DatePickerComponentProps } from './DatePickerComponent.types';
+
+export const DatePickerComponent = forwardRef<
+  HTMLDivElement,
+  DatePickerComponentProps
+>(({ ...rest }: DatePickerComponentProps, fwdRef) => {
+  const { isOpen, menuId, disabled, isSelectOpen } = useDatePickerContext();
+  const {
+    refs,
+    value,
+    closeMenu,
```

Comment:
should closeMenu and isOpen be colocated in the same context?

Reply:
Ideally yes, but the `openMenu` and `closeMenu` functions trigger side effects that will be different between single & range picker, so we can't include them in the shared `useDatePickerContext`. 

If this is something we want to resolve we can either:
1. Move the `isOpen` state from `DatePickerContext` to `SingleDateContext`. (This will mean re-creating this state in `DateRangeContext` as well)
2. Rename the current state var to `_isOpen` and export `isOpen` from `DatePickerContext`

Neither are hard to do, this is just how the contexts ended up working

Reply:
Ah interesting. I'd lean towards option 1, personally